frontend/src/components/App/Layout.tsx

```diff
@@ -156,13 +163,87 @@ export default function Layout({}: LayoutProps) {
           });
 
           if (isConfigDifferent) {
-            dispatch(setConfig(configToStore));
+            dispatch(setConfig({ ...configToStore }));
```

Comment:
There's quite a lot of logic here that should probably be somewhere else.

Reply:
Can you please explain a bit more? Do you mean the part about `statelessClusters` or this whole function. Also to move elsewhere do you mean in the same file or a separate file? Thanks

Reply:
There's still a lot of logic related to this in the Layout component.

Reply:
Got it, updating the same.